src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -424,9 +434,9 @@ ArrayHandler BindArray(NpgsqlTypeMapping elementMapping)
             NpgsqlDbType? arrayNpgsqlDbType = null,
             Type[]? elementClrTypes = null)
         {
-            var arrayHandler = elementHandler.CreateArrayHandler(arrayPgType, _connector.Settings.ArrayNullabilityMode);
 
-            Bind(arrayHandler, arrayPgType, arrayNpgsqlDbType);
+            var arrayHandler = elementHandler.CreateArrayHandler(arrayPgType, _connector.Settings.ArrayNullabilityMode);
+            arrayHandler = GetOrBind(arrayHandler, arrayPgType, arrayNpgsqlDbType);
```

Comment:
Note that we flow down to handle the elementClrTypes even if we didn't bind a new handler (was already there). This isn't really a problem, but it could be prevented by checking the returned instance against the given one (as you do for base types) or by refactoring to use TryAdd (more significant).

Reply:
Yeah I did consider it, could add it...